### ISSUE TITLE
possible mistake in examples/autoencoder.py

### ISSUE THREAD

**Issue by qiuhaolu** [rusty1s/pytorch_geometric#264]
Thank you for the great library! It is very helpful.

I found the results of this example (0.98 AUC and AP) seems too good and much better than the original paper which is around 0.84. So I checked the code and found 

def train():
    model.train()
    optimizer.zero_grad()
    z = model.encode(x, edge_index) 
    loss = model.recon_loss(z, data.train_pos_edge_index)
    if args.model in ['VGAE']:
        loss = loss + 0.001 * model.kl_loss()
    loss.backward()
    optimizer.step()

"z = model.encode(x, edge_index)" seems to input the validation and test data when training. Should it be something like " z = model.encode(x, data.train_pos_edge_index) "?

I am new in this and please let me know if I am wrong. Thank you for your great work.

**Comment by rusty1s** [comment]
Yes, of course! Fixed in master. Thank you very much :)

**Comment by qiuhaolu** [comment]
Thanks. But now the results seem not good enough. It only achieves AUC and AP of around 0.77 for GAE and even worse 0.65 for VGAE. According to the paper, the 2 numbers should be around 0.84 because this script uses the same setting with that paper. I guess there might be some mistake in here?

It would be nice if this example script could reproduce the results of the original paper.

**Comment by rusty1s** [comment]
Which paper do you mean? I do not see any results of 0.84 in the paper.

You need to also update PyG from master (the model changed a bit).

**Comment by qiuhaolu** [comment]
Kipf, Thomas N., and Max Welling. "Variational graph auto-encoders." arXiv preprint arXiv:1611.07308 (2016).

**Comment by rusty1s** [comment]
There is still no result of 0.84 in the [paper](https://arxiv.org/pdf/1611.07308.pdf). I can confirm the results of Cora and PubMed, CiteSeer is a bit lower.

**Comment by qiuhaolu** [comment]
In table 1, GAE* and VGAE*. 
GAE* 84.3 ± 0.02 88.1 ± 0.01 78.7 ± 0.02 84.1 ± 0.02 82.2 ± 0.01 87.4 ± 0.00
VGAE* 84.0 ± 0.02 87.7 ± 0.01 78.9 ± 0.03 84.1 ± 0.02 82.7 ± 0.01 87.5 ± 0.01
GAE 91.0 ± 0.02 92.0 ± 0.03 89.5 ± 0.04 89.9 ± 0.05 96.4 ± 0.00 96.5 ± 0.00
VGAE 91.4 ± 0.01 92.6 ± 0.01 90.8 ± 0.02 92.0 ± 0.02 94.4 ± 0.02 94.7 ± 0.0

**Comment by rusty1s** [comment]
So you are testing without node features?

**Comment by qiuhaolu** [comment]
I think the "x" in your code is the feature matrix, which is n*1433 dim where 1433 is an one-hot representation for each Cora data point. So it seems to me the script is using input features and the results should be around 0.91.

I also tried using identity matrix as features (i.e., x), and the results are worse. This is the condition that does not use node features.

So basically I tried with and without features and the results cannot reach 0.84 (the lowest in this table for Cora).

Please let know if I am doing anything wrong...

**Comment by rusty1s** [comment]
Ok, did you install current master from source? This is absolutely necessary.

**Comment by qiuhaolu** [comment]
No, I used Pip to install pyg last week and my pytorch is 1.1.

**Comment by rusty1s** [comment]
Ok, please download this repo, run `python setup.py develop` inside of it and rerun the example.

**Comment by qiuhaolu** [comment]
I did it and it seems VGAE is doing a little better and achieves around 0.7. GAE is around 0.79 on Cora. Citeseer is worse.

**Comment by rusty1s** [comment]
Are you sure? GAE is around 90-91 to me. VGAE needs more epochs.

**Comment by qiuhaolu** [comment]
I redid everything on another laptop and now the results look good.

I was using Windows Subsystem for Linux (WSL) and maybe this is the problem.

Thank you for your help.

**Comment by rusty1s** [comment]
This may be caused by  double versions of a package. Uninstalling all versions via `pip uninstall` and a single reinstall should fix this.

**Comment by qiuhaolu** [comment]
...you are right. Thanks.
